components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/core/WebAuthnService.java

```diff
@@ -115,31 +121,34 @@
         PublicKeyCredentialCreationOptions credentialCreationOptions = relyingParty
                 .startRegistration(buildStartRegistrationOptions(user));
 
-        RegistrationRequest request = new RegistrationRequest(user.toString(), generateRandom(),
-                credentialCreationOptions);
+        RegistrationRequest request = new RegistrationRequest(generateRandom(), credentialCreationOptions);
 
         FIDO2Cache.getInstance().addToCacheByRequestId(new FIDO2CacheKey(request.getRequestId().getBase64()),
                 new FIDO2CacheEntry(jsonMapper.writeValueAsString(request.getPublicKeyCredentialCreationOptions()),
                         null, originUrl));
         return Either.right(request);
     }
 
-    public void finishRegistration(String challengeResponse) throws FIDO2AuthenticatorException, IOException {
+    public void finishRegistration(String challengeResponse) throws FIDO2AuthenticatorServerException,
+            FIDO2AuthenticatorClientException {
 
         RegistrationResponse response;
         try {
             response = jsonMapper.readValue(challengeResponse, RegistrationResponse.class);
+        } catch (JsonParseException | JsonMappingException e) {
+            throw new FIDO2AuthenticatorClientException("Finish FIDO2 device registration request is invalid.",
```

Comment:
include 'e' also

Reply:
Fixed in https://github.com/wso2-extensions/identity-local-auth-fido/pull/41/files#diff-f01c1476dba8be79bcd0e8f51ffa9a94R141